room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacTypeElement.kt

```diff
@@ -77,6 +74,23 @@ internal sealed class JavacTypeElement(
     }
 
     override fun isKotlinObject() = kotlinMetadata?.isObject() == true
+    override fun isCompanionObject() = kotlinMetadata?.isCompanionObject() == true
+    override fun isDataClass() = kotlinMetadata?.isDataClass() == true
+    override fun isValueClass() = kotlinMetadata?.isValueClass() == true
+    override fun isFunctionalInterface() = kotlinMetadata?.isFunctionalInterface() == true
+    override fun isExpect() = kotlinMetadata?.isExpect() == true
+
+    override fun isAnnotationClass(): Boolean {
+        return kotlinMetadata?.isAnnotationClass() ?: (element.kind == ElementKind.ANNOTATION_TYPE)
```

Comment:
minor nit, probably better to do the elements check first but feel free to keep it this way.

Reply:
I can switch it, seems faster